harness/harness-core-ui
# fix: [CDS-83171]: handle the pipeline updated pop-up appearing for sync enabled pipelines

### Summary

- When bi-directional sync is enabled we need to use the pipeline from the API to render the pipeline studio on reload to sync the latest git changes
- We do not want to show unsaved changes and pipeline is updated by some other user in such cases 

Test Cases Executed :- 

- Create a remote pipeline , update on git and reload the pipeline , the changes appears
- Open the same remote pipeline in multiple tabs and soft fetch should function fine and on reload changes from git should appear
- soft fetch and reload working fine for inline pipelines.


#### Screenshots


https://github.com/harness/harness-core-ui/assets/106532291/373d6d7b-0513-481e-ba08-ab6ad43333b4



#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

#### [Contributor license agreement](https://github.com/harness/harness-core-ui/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)
